pandas/core/dtypes/dtypes.py

```diff
@@ -240,6 +241,77 @@ def _from_categorical_dtype(cls, dtype, categories=None, ordered=None):
             ordered = dtype.ordered
         return cls(categories, ordered)
 
+    @classmethod
+    def _from_values_or_dtype(cls, values=None, categories=None, ordered=None,
+                              dtype=None):
+        """
+        Construct from the inputs used in :class:`Categorical` construction.
```

Comment:
can you change this verbiage a bit. The first sentence is obsolete, this just constructs the dtype.

You an say it doesn't factorize, but its not an 'internal helper method' its a constructor for the dtype.